src/flp/types.rs

```diff
@@ -486,14 +491,21 @@ where
             )));
         }
 
-        let max = F::Integer::from(F::one());
-        for value in measurement {
-            if *value > max {
-                return Err(FlpError::Encode("Count value must be 0 or 1".to_string()));
+        let max = F::Integer::try_from(1 << self.bits).unwrap();
+        let mut flattened = Vec::with_capacity(self.bits * self.len);
```

Comment:
I think we should check whether `1 << self.bits` or `self.bits * self.len` overflows and return an error if so. `SumVec::new` is probably a good place to do that, since if you check it there, you can assume those values are sane everywhere else in the `impl`.